include/AudioEngineProfiler.h

```diff
@@ -53,11 +54,31 @@ class AudioEngineProfiler
 
 	void setOutputFile( const QString& outputFile );
 
+	enum class DetailType {
+		NoteSetup,
+		Instruments,
+		Effects,
+		Mixing,
+		DetailCount
+	};
+
+	void startDetail(const DetailType type) { m_detailTimer[static_cast<int>(type)].reset(); }
+	void finishDetail(const DetailType type)
+	{
+		m_detailTime[static_cast<int>(type)] = m_detailTimer[static_cast<int>(type)].elapsed();
+	}
+
+	int detailLoad(const DetailType type) const { return m_detailLoad[static_cast<int>(type)]; }
 
 private:
 	MicroTimer m_periodTimer;
-	int m_cpuLoad;
+	float m_cpuLoad;
 	QFile m_outputFile;
+
+	// Use arrays to avoid dynamic allocations in realtime code
+	std::array<MicroTimer, static_cast<int>(DetailType::DetailCount)> m_detailTimer;
+	std::array<int, static_cast<int>(DetailType::DetailCount)> m_detailTime{0};
+	std::array<float, static_cast<int>(DetailType::DetailCount)> m_detailLoad{0};
```

Comment:
It would be more appropriate to cast the size to `std::size_t`, since that is the actual template parameter type here. Also, since the `DetailCount` member is only ever going to be used as an integer, and not as an enumerator, I think it makes sense to define it as such outside the enum definition. In #6760, I standardised this pattern as the following:
> Final enumerators representing the total number of enumerators have all been renamed to `Count`. Any that were used numerically have had a `std::size_t` constant with the original enumerator name added alongside the enum.

In this case, you'd have:
```cpp
enum class DetailType {
	// ...
	Mixing,
	Count
};
constexpr static auto DetailCount = static_cast<std::size_t>(DetailType::Count);

// ...

std::array<MicroTimer, DetailCount> m_detailTimer;
```

Reply:
Thanks, that will be much cleaner.